kotlinx-coroutines-core/jsAndWasmJsShared/src/internal/CoroutineExceptionHandlerImpl.kt

```diff
@@ -0,0 +1,14 @@
+package kotlinx.coroutines.internal
+
+import kotlinx.coroutines.*
+
+internal expect interface JsAny
+
+internal expect fun Throwable.toJsException(): JsAny
+
+internal fun throwAsync(e: JsAny): Unit = js("setTimeout(function () { throw e }, 0)")
```

Comment:
Please add a comment explaining the principle behind this solution. It looks mysterious: why set something to run after a timeout of *zero*?